File: 프로그래머스/0/120813. 짝수는 싫어요/짝수는 싫어요.java
class Solution {
    public int[] solution(int n) {
        int[] answer;
        int index = 0;  // 배열 answer 배열 크기
        int a = 0;      // 배열 answer 인덱스 
        
        // 배열 answer 크기 구하기
        for(int i=1; i<=n; i+=2) {
            index += 1;
        }
        
        // 배열 answer 초기화
        answer = new int[index];
        
        // 홀수가 오름차순으로 담긴 배열 생성
        for(int i=1; i<=n; i+=2) {
            answer[a++] = i;
        }
        
        return answer;
    }
}

File: 프로그래머스/0/120813. 짝수는 싫어요/짝수는 싫어요.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int* solution(int n) {
    // return 값은 malloc 등 동적 할당을 사용해주세요. 할당 길이는 상황에 맞게 변경해주세요.
    
    int num = 0;
    for(int i=1; i<=n; i++) {
        if(i % 2 == 1) num += 1;
    }
    
    int* answer = (int*)malloc(sizeof(int)*num);
    int k = 0;
    for(int i=1; i<=n; i++) {
        if(i % 2 == 1) answer[k++] = i;
    }
    return answer;
}